pkg/kubeapiserver/admission/configuration/configuration_manager.go

```diff
@@ -81,7 +94,30 @@ func (a *poller) notReady() {
 	a.ready = false
 }
 
+func (a *poller) bootstrapping() {
+	var ready bool
+	// bootstrapRetries is read-only, so no lock required.
+	for i := 0; i < a.bootstrapRetries; i++ {
```

Comment:
what if instead of doing this you made a `bootstrapped chan struct{}` and then whenever ready is set to true, do `if !closed(a.bootstrapped) { close(a.bootstrapped) }` and then to wait for it you can just do `<-bootstrapped`.